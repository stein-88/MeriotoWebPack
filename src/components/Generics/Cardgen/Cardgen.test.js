import React from 'react'
import renderer from 'react-test-renderer'
import CardGen from './CardGen'

describe('Test on CardGen Component', () => {
    const compCardGen = renderer.create(<CardGen />)
    test('Should render without props', () => {
        const jsonCardGen = compCardGen.toJSON()
        expect(jsonCardGen).toMatchSnapshot()
    })
    test('Should have an IMG only', () => {
        const expectResult = compCardGen.root.findAllByProps({ className: "img-fluid sizeimg" })
        expect(expectResult).toHaveLength(1)
    })
    test('Should have an cursorPointer when we pass the image', () => {
        const compCardGen = renderer.create(<CardGen theimage={'linklaq'} />)
        const expectResult = compCardGen.root.findAllByProps({ className: "img-fluid sizeimg cursorPointer" })
        expect(expectResult).toHaveLength(1)
    })
    test('Should have 1 title and 1 subtitle', () => {
        const compCardGen = renderer.create(<CardGen thetitle={'tituloaq'} thesubtitle={'subtituloaq'} />)
        const expectTitle = compCardGen.root.findAllByProps({ className: "h4 pt-2" })
        const expectSubtitle = compCardGen.root.findAllByProps({ className: "h5 pt-2" })
        expect(expectTitle).toHaveLength(1)
        expect(expectSubtitle).toHaveLength(1)
    })
    test('Should have 2 buttons', () => {
        const compCardGen = renderer.create(<CardGen botao={[
            {
                text: 'text1',
                onClick: () => null
            },
            {
                text: 'text2',
                onClick: () => null
            }
        ]} />)
        const expectResult = compCardGen.root.findAllByProps({ ['data-test']: "botao" })
        expect(expectResult).toHaveLength(2)
    })
})