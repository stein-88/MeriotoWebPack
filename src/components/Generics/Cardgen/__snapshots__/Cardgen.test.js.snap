// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on CardGen Component Should render without props 1`] = `
<section
  className="w-100 my-3"
>
  <section
    className="w-100 pb-3 card"
  >
    <section
      className="w-100 pb-3"
    >
      <img
        alt="can be anything"
        aria-hidden="true"
        className="img-fluid sizeimg"
        onClick={[Function]}
        src="test-file-stub"
      />
    </section>
    <section
      className="w-100"
    />
  </section>
</section>
`;
